### PR TITLE
libvmi: Promote libvmi from e2e usages to the root `pkg/`

### PR DESCRIPTION
**Pull request by EdDev**
### What this PR does

`libvmi` has been widely used by source code outside e2e tests. As it serves now both unit tests and e2e tests, it fits to exist in a location that fits both parties.

By promoting `libvmi` at the root `pkg/`, we allow both unit and e2e tests to use its services.

While all libvmi builders moved, the factory was left in the e2e test. The factory package has been renamed to `libvmifact` and used solely in the e2e tests per need.

Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/libvmi/architecture.go [hunk 1]
```diff
@@ -25,6 +25,6 @@ func RegisterArchitecture(arch string) {
 	architecture = arch
 }
 
-func isARM64() bool {
+func IsARM64() bool {
```
Comment (fossedihelm) [inline pkg/libvmi/architecture.go:28]: Non sono sicuro che l'architettura debba essere spostata con libvmi.
Forse dovrebbe rimanere con factory

Comment (EdDev) [reply pkg/libvmi/architecture.go:28]: You are right.

Done

**Comment by EdDev** [comment]
[change](https://github.com/kubevirt/kubevirt/compare/8622d8df7203ba19eb430999eaf76409e1e97c91..452c404f65fbc0be5be4d27e223f292f30342964): Keep the architecture portion under the e2e tests. As a response to the review.

**Comment by iholder101** [comment]
Thank you @EdDev, great seeing this! Awesome and important job! :tada: 

> While all libvmi builders moved, the factory was left in the e2e test. The factory package has been renamed to libvmifact and used solely in the e2e tests per need.

I think I get where you're coming from when you say this - since we do tweak these factories according to our e2e tests needs. I have two concerns regarding this:
1. In [another PR](https://github.com/kubevirt/kubevirt/pull/11494) I got some pushback regarding using `tests/` code in unit tests (which makes a production package import a test package). I think that the factories are very helpful for unit tests, and it would be a shame to duplicate the factory code all over them.
2. At least IMO the end-goal is that Kubevirt users would be able to use libvmi for their own needs. I think that providing some default factories to serve as a reference could be very valuable for such users that are not familiar yet with libvmi.

WDYT about moving the factories outside of the tests package as well?
We can document that these are just reference factories that we use at our testing and serve as an example. If in the future we'll get to a disagreement over this we can always fallback to the approach presented in this PR and bring factories back to tests package.

**Comment by EdDev** [comment]
> Thank you @EdDev, great seeing this! Awesome and important job! 🎉
> 
> > While all libvmi builders moved, the factory was left in the e2e test. The factory package has been renamed to libvmifact and used solely in the e2e tests per need.
> 
> I think I get where you're coming from when you say this - since we do tweak these factories according to our e2e tests needs. I have two concerns regarding this:
> 
>     1. In [another PR](https://github.com/kubevirt/kubevirt/pull/11494) I got some pushback regarding using `tests/` code in unit tests (which makes a production package import a test package). I think that the factories are very helpful for unit tests, and it would be a shame to duplicate the factory code all over them.

e2e factories should not be needed by unit tests. I hope it is clear that if they are used in unit tests, someone tested something much bigger than an unit.

But if some specific unit tests do need a factory, one should be created for them specifically.

> 
>     2. At least IMO the end-goal is that Kubevirt users would be able to use libvmi for their own needs. I think that providing some default factories to serve as a reference could be very valuable for such users that are not familiar yet with libvmi.

The current factories are serving only the e2e tests.
There is a need to convince that a factory makes sense to a wider common usage, e.g. for e2e tests, unit tests and potential external projects.

Usually it is easier to convince after you have actual individual factories that can be promoted to the central library/package.
This is exactly what happened with libvmi itself.

> WDYT about moving the factories outside of the tests package as well? We can document that these are just reference factories that we use at our testing and serve as an example. If in the future we'll get to a disagreement over this we can always fallback to the approach presented in this PR and bring factories back to tests package.

IMO we should promote only common and well reasoned API/s. To document something we have the README which one can add examples to if it makes sense.

**Comment by iholder101** [comment]
> e2e factories should not be needed by unit tests.

Yeah, rethinking about this I guess you're right. There isn't an obvious need to use factories in unit tests.

Alrighty, let's start off this way. We can always circle back to discussing which factories make sense for a broader audience.

Thanks again for this important PR.
/approve

**Review by fossedihelm — APPROVED** [review]
Thanks!

**Pull request merged**